doc/technotes/errorHandling.rst

```diff
@@ -0,0 +1,190 @@
+.. _readme-errorHandling:
+
+==============
+Error Handling
+==============
+
+This README describes the error handling feature as currently implemented.
+
+
+Errors
+------
+A base class ``Error`` is defined in the Chapel standard modules.
+
+::
+
+  class Error {
+    var msg: string;
+
+    proc Error(_msg: string) {
+      msg = _msg;
+    }
+  }
+
+Users may use the provided ``Error`` directly, or create their own error
+hierarchy with ``Error`` at its root.
+
+::
+
+  class MyError: Error {}
+
+  class MyIntError: MyError {
+    var i: int;
+  }
+
+Chapel does not provide a standard set of ``Error`` classes.
+
+
+Throwing Errors
+---------------
+``throw`` statements throw errors from a function to its callee.
+
+::
+
+  throw new Error("with a message");
```

Comment:
While `"with a message"` is nice in that the above reads like a sentence, it doesn't match how you'd actually use it, since "with a message" is a pretty poor error message. Let's just put something like "error message here" in the string.